sys/auto_init/auto_init.c

```diff
@@ -639,4 +639,10 @@ void auto_init(void)
     extern void dhcpv6_client_auto_init(void);
     dhcpv6_client_auto_init();
 #endif /* MODULE_AUTO_INIT_DHCPV6_CLIENT */
+
+    if (IS_USED(MODULE_GNRC_DHCPV6_CLIENT_6LBR)) {
```

Comment:
> Since the rest of the file uses `#ifdef`, I'd suggest to also use it here so it follows the same pattern.

First step for going a different direction? I can change it if you insist

> Also shouldn't his be mutually exclusive with `MODULE_AUTO_INIT_DHCPV6_CLIENT`?

It isn't necessarily mutually exclusive. The latter initialized will just take over the `sock`. I'll add a `LOG_ERROR()` for that case.

Reply:
> It isn't necessarily mutually exclusive. The latter initialized will just take over the `sock`. I'll add a `LOG_ERROR()` for that case.

Or better: a compile time error!

Reply:
See 0f3670a44d

Reply:
> First step for going a different direction? I can change it if you insist

I'd say we should convert them all or none.
Having a mish-mash in the same file/function is just confusing.

Reply:
I won't do it in this PR. But mish-mash is just a side-effect of a transition period: Even if we open a PR that converts all of them now, older PRs that are merged later and use the `#ifdef` variant, would still cause inconsistencies (given that both reviewer and contributor don't know about the earlier PR).

Reply:
So what's your verdict now? Keep it as is or use `#ifdef`?

Reply:
I think it would be cleaner to use `#ifdef` now and have a separate PR to convert it all to `IS_USED`  - other PRs would get a merge conflict that they'd have to resolve.

But is `#if IS_USED()` always preferred over `#ifdef`? In this case I don't see a benefit for using the macro version.

Reply:
The main benefit of `if (IS_USED(…))` (not `#if IS_USED(…)`) that at compile-time the branch is still checked by the compiler and then only thrown out due to optimization.

Reply:
Went for `#ifdef` for now.